### ISSUE TITLE
Improve HomeSecurity

### ISSUE THREAD

**Issue by lipkam** [justonhart/Thoughtwave#42]
Improve on the homeSecurity method in the RoomManagement class. If there are multiple enemy creeps or they are strong ones then spawn 2 units. One can be the current attacker, but another one should be a healer. 

Possibilities:

- Make both units go to the enemy together (healer in the back)
- If you spawn ranged unit maybe leave healer outside of enemy range and just heal the ranged unit (attacking over the wall)

Basically, we need some type of "threat assessment" method, a corresponding spawning logic and then finally a way for the creeps to attack as a unit.

**Comment by justonhart** [comment]
We should track something like "threat level" for our rooms. Based off of this value, we can have different spawn priorities or room behaviors. Examples include:
- spawning extra distributors or workers to help refill and repair
- stop spawning miners to prevent energy waste on creeps that would die
- request assistance from nearby rooms
